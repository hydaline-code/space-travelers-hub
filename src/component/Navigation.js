import React from 'react';
import { NavLink } from 'react-router-dom';
import './styles/Navigation.css';

function Navigation() {
  return (
    <div>
      <div className="navbar">
        <div className="logo">
          <img src="planet.png" style={{ width: '50px', height: '50px' }} alt="" />
          <h1>Space Travelers&apos; Hub</h1>
        </div>
        <div className="nav-links" />
        <NavLink
          to="/Rockets"
          className="view"
          style={({ isActive }) => ({
            textDecoration: isActive ? 'underline' : '',
            color: isActive ? 'blue' : '',
          })}
        >
          Rockets
        </NavLink>
        <NavLink
          to="/Missions"
          className="view"
          style={({ isActive }) => ({
            textDecoration: isActive ? 'underline' : '',
            color: isActive ? 'blue' : '',
          })}
        >
          Missions
        </NavLink>
        <NavLink
          to="/Dragons"
          className="view"
          style={({ isActive }) => ({
            textDecoration: isActive ? 'underline' : '',
            color: isActive ? 'blue' : '',
          })}
        >
          Dragons
        </NavLink>
        <NavLink
          to="/Profile"
          className="My-profile"
          style={({ isActive }) => ({
            textDecoration: isActive ? 'underline' : '',
            color: isActive ? 'blue' : '',
          })}
        >
          My Profile
        </NavLink>
      </div>
    </div>
  );
}
export default Navigation;
