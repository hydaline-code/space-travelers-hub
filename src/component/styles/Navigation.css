@import url('https://fonts.googleapis.com/css2?family=Gothic+A1:wght@100;200;300;400;500;600;700;800;900&family=Lato:wght@100;300;400;700&family=Montserrat:wght@100;200;300;400;500;600&family=Permanent+Marker&family=Poppins:ital,wght@0,100;0,300;0,400;0,500;0,600;1,100&family=Space+Grotesk:wght@300;400;500;600;700&family=Titan+One&display=swap');

body {
  font-family: 'Poppins', sans-serif;
}

.navbar {
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid rgba(111, 111, 111, 0.6);
  width: 85%;
  margin: 0 auto;
}


.logo {
  display: flex;
  justify-content: space-around;
  width: 350px;
}

.nav-links {
  display: flex;
  justify-content: flex-end;
}

.nav-links a {
  margin-left: 1rem;
  text-decoration: none;
  font-size: 20px;
  color: #0290ff;
}

a:hover {
  cursor: pointer;
}

.My-profile {
  border-left: 3px solid rgba(111, 111, 111, 0.6);
  padding-left: 1rem;
}

.logo img {
  width: 50px;
  height: 50px;
  margin-right: 10px;
}

.logo h1 {
  color: rgb(19, 18, 18);
  font-size: 26px;
}

.view:hover, .My-profile:hover {
  text-decoration: underline;
  color: blue;
}
