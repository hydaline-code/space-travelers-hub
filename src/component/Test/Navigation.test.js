import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '../Navigation';

describe('Navigation Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navigation />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
