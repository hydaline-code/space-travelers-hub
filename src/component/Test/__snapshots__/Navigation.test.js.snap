// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component matches the snapshot 1`] = `
<div>
  <div
    className="navbar"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="planet.png"
        style={
          Object {
            "height": "50px",
            "width": "50px",
          }
        }
      />
      <h1>
        Space Travelers' Hub
      </h1>
    </div>
    <div
      className="nav-links"
    >
      <a
        className="view"
        href="/Rockets"
        onClick={[Function]}
        style={
          Object {
            "color": "",
            "textDecoration": "",
          }
        }
      >
        Rockets
      </a>
      <a
        className="view"
        href="/Missions"
        onClick={[Function]}
        style={
          Object {
            "color": "",
            "textDecoration": "",
          }
        }
      >
        Missions
      </a>
      <a
        className="view"
        href="/Dragons"
        onClick={[Function]}
        style={
          Object {
            "color": "",
            "textDecoration": "",
          }
        }
      >
        Dragons
      </a>
      <a
        className="My-profile"
        href="/Profile"
        onClick={[Function]}
        style={
          Object {
            "color": "",
            "textDecoration": "",
          }
        }
      >
        My Profile
      </a>
    </div>
  </div>
</div>
`;
